cli/archive.go

```diff
@@ -0,0 +1,114 @@
+package cli
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"io/fs"
+	"math"
+	"os"
+	"path/filepath"
+
+	"go.viam.com/utils"
+	"golang.org/x/exp/maps"
+)
+
+// getArchiveFilePaths traverses the provided rootpaths recursively,
+// collecting the file paths of all regular files and returns them in a slice.
+// This list of paths should be passed to createArchive.
+func getArchiveFilePaths(rootpaths []string) ([]string, error) {
+	files := map[string]bool{}
+	for _, pathRoot := range rootpaths {
+		err := filepath.WalkDir(filepath.Clean(pathRoot), func(path string, info fs.DirEntry, err error) error {
+			if err != nil {
+				return err
+			}
+			if info.Type().IsRegular() {
+				files[path] = true
```

Comment:
why assemble a map + convert to slice? deduping?

Reply:
Yup, deduping. This isn't the prettiest pattern but it is used in lots of places throughout our codebase. If there is a better pattern happy to swap to that.